safe_transaction_service/history/services/balance_service.py

```diff
@@ -204,13 +240,17 @@ def _get_balances(
                 raw_balances.extend(balances[1:] if raw_balances else balances)
 
             # Return ether balance if there are no tokens
-            if not raw_balances:
+            if not erc20_addresses_paginated:
                 raw_balances = self.ethereum_client.erc20.get_balances(safe_address, [])
             # First element should be the ether transfer
         except (IOError, ValueError) as exc:
             raise NodeConnectionException from exc
 
         balances = []
+        if offset != 0 and raw_balances:
```

Comment:
I understand why you do this, but it's a bit tricky. Maybe in the future, we can change `get_balances` method and retrieve erc20 and ether separately. WDYT?

Reply:
Agree I created an issue to handle it https://github.com/safe-global/safe-eth-py/issues/1199